surefire-providers/common-junit48/src/main/java/org/apache/maven/surefire/common/junit48/GroupMatcherCategoryFilter.java

```diff
@@ -63,33 +63,50 @@
     @Override
     public boolean shouldRun( Description description )
     {
-        if ( description.getMethodName() == null || description.getTestClass() == null )
-        {
+        if(invalidTestClass(description)){
```

Comment:
you did not run the build locally otherwise the Maven Checkstyle Plugin wouls top you with notices regarding code style. Pls run `mvn install -P run-its`.

Reply:
yes, I didn't use the Apache formatter. I'll fix it